flink-core/src/main/java/org/apache/flink/configuration/BatchExecutionOptions.java

```diff
@@ -32,4 +37,106 @@ public class BatchExecutionOptions {
+                                            .build());
+
+    @Documentation.Section({Documentation.Sections.EXPERT_SCHEDULING})
+    public static final ConfigOption<Integer> ADAPTIVE_AUTO_PARALLELISM_DEFAULT_SOURCE_PARALLELISM =
+            key("execution.batch.adaptive.auto-parallelism.default-source-parallelism")
+                    .intType()
+                    .defaultValue(1)
+                    .withDeprecatedKeys(
+                            "jobmanager.adaptive-batch-scheduler.default-source-parallelism")
+                    .withDescription(
+                            Description.builder()
+                                    .text(
+                                            "The default parallelism of source vertices if %s has been set to %s",
+                                            code(SCHEDULER.key()),
+                                            code(
+                                                    JobManagerOptions.SchedulerType.AdaptiveBatch
+                                                            .name()))
+                                    .build());
+
+    @Documentation.Section({Documentation.Sections.EXPERT_SCHEDULING})
+    public static final ConfigOption<Boolean> SPECULATIVE_ENABLED =
+            key("execution.batch.adaptive.speculative.enabled")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDeprecatedKeys("jobmanager.adaptive-batch-scheduler.speculative.enabled")
+                    .withDescription("Controls whether to enable speculative execution.");
+
+    @Documentation.Section({Documentation.Sections.EXPERT_SCHEDULING})
+    public static final ConfigOption<Integer> SPECULATIVE_MAX_CONCURRENT_EXECUTIONS =
+            key("execution.batch.speculative.max-concurrent-executions")
+                    .intType()
+                    .defaultValue(2)
+                    .withDeprecatedKeys(
+                            "jobmanager.adaptive-batch-scheduler.speculative.max-concurrent-executions")
+                    .withDescription(
+                            "Controls the maximum number of execution attempts of each operator "
+                                    + "that can execute concurrently, including the original one "
+                                    + "and speculative ones.");
+
+    @Documentation.Section({Documentation.Sections.EXPERT_SCHEDULING})
+    public static final ConfigOption<Duration> BLOCK_SLOW_NODE_DURATION =
+            key("execution.batch.speculative.block-slow-node-duration")
+                    .durationType()
+                    .defaultValue(Duration.ofMinutes(1))
+                    .withDeprecatedKeys(
+                            "jobmanager.adaptive-batch-scheduler.speculative.block-slow-node-duration")
+                    .withDescription(
+                            "Controls how long an detected slow node should be blocked for.");
 }
```

Comment:
A private constructor should be added. A reference can be `RestartStrategyOptions`.